Runtime/Scripts/RGBotServerListener.cs

```diff
@@ -558,9 +558,20 @@ private void FixedUpdate()
                         }
                     }
                 }
-                totalState[state["id"].ToString()] = state;
+                
+                var key = gameObjectState.id.ToString();
+                if (fullGameState.TryGetValue(key, out var combinedGameObjectState))
+                {
+                    // if GameObject has multiple state scripts attached to it, then combine fields into one RGStateEntity
+                    gameObjectState.ToList().ForEach(x => combinedGameObjectState[x.Key] = x.Value);
```

Comment:
Do you have an example of what this ends up looking like? From my reading, it seems like a random agent key/id is selected, and under that the other ids are nested? I am thinking I maybe be misunderstanding...

Reply:
for each component with RGState...
  * if entity hasn't been added to the full state object yet then add all state info
  * else append state info to existing entity in the full state object. Any key conflicts will be overwritten.

Ex. If I have one state script that defines `isIdle` and another state script that defines `inventory` then we end up with what we've been expecting:
```
<entity id>: {
  "isPlayer": true,
  "isRuntimeObject": true,
  "position": ... , 
  "rotation": ... ,
  "isIdle": true,
  "inventory": [...] ,
  // other state stuff
}
```

Reply:
Got it, thank you!